app/controllers/contracts_controller.rb

```diff
@@ -17,6 +17,67 @@ def index
     
   end
 
+  def accept
+    @contract = Contract.find(params[:contract_id])
+    @contract.update(is_accepted: true)
+    @contract.update(status: :approved)
+    redirect_to root_path
+  end
+
+  def reject
+    @contract = Contract.find(params[:contract_id])
+
+    if current_user == @contract.beneficiary
+      @contract.destroy
+    end
+    
+    respond_to do |format|
+      format.html { redirect_to root_path, notice: "Contract was rejected." }
+      format.json { head :no_content }
+    end
+  end
+
+  def modify
+    @contract = Contract.find(params[:contract_id])
+    @contract.update(status: :modification_requested)
+
+    respond_to do |format|
+      format.html { redirect_to root_path, notice: "Modification requested to #{@contract.owner.name}" }
+      format.json { head :no_content }
+    end
+  end
+
+  def accept_modification
+    @contract = Contract.find(params[:contract_id])
```

Comment:
you can remove this line if you add `accept_modification` in the array at line 2